infrastructure/resource_group_level/modules/redis.bicep

```diff
@@ -0,0 +1,42 @@
+param location string
+param environment string
+
+var redisName = 'tfs-${environment}-redis'
+
+// See https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-redis-cache-bicep-provision?tabs=CLI
+// for an example Bicep file
+
+// TODO:FN-504 Configure private link.
+// See line 718 in developmentinfrastructure.yml
+// and https://docs.microsoft.com/en-us/azure/azure-cache-for-redis/cache-private-link
+
+// TODO:FN-504 decide what sku to use.
+// Note that it isn't recommended to use Basic or C0 in production
+// See https://learn.microsoft.com/en-gb/azure/azure-cache-for-redis/cache-best-practices-development
+var sku = {
+  name: 'Basic'
+  family: 'C'
+  capacity: 0
+}
+
+resource redis 'Microsoft.Cache/redis@2022-06-01' = {
+  name: redisName
+  location: location
+  tags: {
+    Environment: 'Preproduction'
+  }
+  properties: {
+    redisVersion: '6.0'
+    sku: sku
+    enableNonSslPort: false
+    minimumTlsVersion: '1.2'
```

Comment:
small comment here: can we put some of these values in a constant/variable at the top? it should be easier to change from there if we need to in the future

Reply:
As I've just added above - I'll make `sku` a parameter. I was going to just make it an object, but I note that the example splits them so it can put validation on them. If you prefer that approach, say and I'll use it. https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-redis-cache-bicep-provision?tabs=CLI

I don't plan on adding any of the other parameters here as I don't think they will be changed.